### ISSUE TITLE
Add YTMusic search function

### ISSUE THREAD

**Issue by DDWIM** [stephane-r/Youtube-Audio-Player#161]
Actual music titels are not listed on YouTube. They can only be found via YTMusic.
As they are ordinary YT videos, they can be played back via Invidious over the videoID.

To be able to easily find music like on Spotify f.e., I would suggest a function to optionally search via YTMusic. As there are multiple YTMusic APIs available on Github, it shouldn´t be too much work to implement this.

**Comment by stephane-r** [comment]
Hey @DDWIM 

I've never used Youtube Music. If I'm mistaken, you need to be a premium member on Youtube, right ?

**Comment by DDWIM** [comment]
No, it´s basically just a different front-end for YouTube, where you can find the on YouTube unlisted songs.

If you go to [https://music.youtube.com/](https://music.youtube.com/) and search for a title, you find the actual music under "Songs". If you click on it, basically a YouTube player in a different design opens and plays audio only.

As this is still a normal YouTube-video, you can view it via the videoID on youtube, simply by removing "music." from the url. Most of the videos will show as "not listed", obviously because Google doesn´t want them to be found over YouTube.
Playing the videos over Invidous works also the same way (over the videoID), just replace the "youtube.com" with an Invidious instance domain.

In my experience these videos don´t play back on every Invidious instance, but the ones with anti-captcha (f.e. [https://invidious.namazso.eu](https://invidious.namazso.eu)) are working really well.

Premium is only needed for ad-free listening, which doesn´t occur with Invidious in the first place 😁

**Comment by stephane-r** [comment]
Ok nice, thank you for this idea. I'll be looking to implement it soon 👍
